### PR TITLE
Bring Pirates of the FSO Back by

### PR DESCRIPTION
**Pull request by FSCyborg**
Preventing a nullptr deep within shipfx_warpin_start()

### CODE REVIEW

**Comment by Goober5000** [comment]
Link to issue #2151.

**Comment by FSCyborg** [comment]
Fantastic point.  Will do.

**Comment by FSCyborg** [comment]
I'm not 100% sure if this is the correct way to check for an uninitialized pointer, because I'm not sure it is the same as nullptr.

File: code/ship/shipfx.cpp [hunk 1]
```diff
@@ -707,6 +708,7 @@ void shipfx_warpout_start( object *objp )
 		return;
 	}
 
+	Assertion(shipp->warpin_effect != nullptr,"shipfx_warpout_start() was fed a ship with an uninitialized warpin_effect.");
```
Comment (Goober5000) [inline code/ship/shipfx.cpp]: You need to check for `warpout_effect` here, not `warpin_effect`.  Also, as an extremely minor nitpick, put a space between the comma and the quote.

**Comment by Goober5000** [comment]
An uninitialized pointer is not the same as nullptr, but the pointers should not be uninitialized here.  The `ship_clear()` function sets them to nullptr.

File: code/ship/shipfx.cpp (hunk 1 above)
Comment (FSCyborg) [reply code/ship/shipfx.cpp]: Thanks!  Should be good now.

File: code/ship/shipfx.cpp [hunk 2]
```diff
@@ -707,6 +708,7 @@ void shipfx_warpout_start( object *objp )
 		return;
 	}
 
+	Assertion(shipp->warpout_effect != nullptr, "shipfx_warpout_start() was fed a ship with an uninitialized warpin_effect.");
```
Comment (Goober5000) [inline code/ship/shipfx.cpp]: It still says warpin_effect at the end of the message.

**Comment by Goober5000** [comment]
It's generally not possible to check for an uninitialized variable.  But debug builds will set them to a known, invalid, but "magic number" value such as `0xcccccccc` to help diagnose problems in the case of a crash.  There's no such help in release builds though.

**Comment by FSCyborg** [comment]
Ok.  Just so you know, any commit I make between 4 and 7 on a weekday I am probably more focused on keeping my son away from the keyboard while I type, than what I am typing....

**Comment by FSCyborg** [comment]
And now??

**Pull request merged**